File: backend/src/main/java/tko/model/dto/workout/WorkoutDTO.java
package tko.model.dto.workout;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class WorkoutDTO {
    @NotNull
    private Long id;
    @NotBlank
    private String description;
    @NotBlank
    private String name;
    @NotBlank
    private String difficult;
}

File: android/app/src/main/java/tk/ssau/fitnesstko/model/dto/WorkoutDto.kt
package tk.ssau.fitnesstko.model.dto

/**
 * Полное представление тренировки.
 * @param id Уникальный идентификатор тренировки (может быть null для новых объектов).
 * @param description Описание тренировки.
 * @param name Название тренировки (например, "Силовая программа для новичков").
 * @param difficult Уровень сложности. Допустимые значения: "EASY", "MEDIUM", "HARD".
 * @param likeCount Количество лайков. Может быть null, если данные не загружены.
 */
data class WorkoutDto(
    val id: Long?,
    val description: String,
    val name: String,
    val difficult: String,
    val likeCount: Int?
)
